### PR TITLE
feat(app): show live labware offset while jogging during Labware Position Check

### LINKED ISSUE #9796: LPC enhancements: Apply stored offsets when selecting "run again"
# Overview
If a user chooses to "run again" from the Recent Protocol Runs table on the Device Details page, we should not prompt them with the "Apply Stored Labware Offset Data" modal, and the labware offsets from that historical run should automatically be applied to the new run. 

# Design
Designs for this short flow live here: 
https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=7792%3A239891

# Implementation Details 
- The labware offsets of the recent runs should live in an accordion inside the Recent Protocol Runs table
- We should show a warning banner in that accordion if the store labware offsets are older than the current robot calibrations

# Acceptance Criteria
- [ ] The "Apply Stored Labware Offset Data" modal should not be shown when the user chooses to "run again" from the Recent Protocol Runs table
- [ ] The labware offsets from that previous run should appear on the labware on the deck map inside the Labware Setup step on the newly create Protocol Run Details page

### LINKED ISSUE #9797: LPC enhancements: LPC workflow updates
# Overview
Inside the Labware Position Check workflow, we want to have the robot move directly to the position of any saved offsets if they're present (instead of 0, 0, 0). We should also show those values in the UI, so the user can easily see what they are and can watch how they change as the pipette is jogged by the user. 

# Design
The UI updates live here: 
https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=7792%3A239891

# Implementation Details 
- If no stored offsets exist for the labware, it should start at the (0, 0, 0) position. 
- If stored offsets do exist, the pipette tip should move to that position and those offsets should be shown in the LPC workflow modal dialog. 

# Acceptance Criteria

- [ ] Stored offsets are shown above the jog controls

- [ ] The offsets should update in the UI as the user jogs the pipette

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

In order to support checking the position of existing offsets that have been applied from historical
runs, this updates the labware position check user flow in two ways.  The first ensures that labware
offsets are not cleared out when LPC is launched. The second addition displays to users the
effective offset so that they can be aware of existing adjustments, and see how their jogging
effects the ultimate value.

Closes https://github.com/Opentrons/opentrons/issues/9796, Closes https://github.com/Opentrons/opentrons/issues/9797

TODO:
- [x] ~unit tests for new `useLabwareOffsetForLabware` hook~

# Changelog

# Review requests

# Risk assessment

### CODE REVIEW

File: api/src/opentrons/protocol_engine/execution/movement.py [hunk 1]
```diff
@@ -136,6 +144,19 @@ async def move_relative(
         except HardwareMustHomeError as e:
             raise MustHomeError(str(e)) from e
 
+        try:
+            point = await self._hardware_api.gantry_position(
+                mount=hw_mount,
+                critical_point=pip_cp,
+                fail_on_not_homed=True,
+            )
```
Comment (mcous) [inline api/src/opentrons/protocol_engine/execution/movement.py:148]: I would move this into the same `try` block as above

File: api/tests/opentrons/protocol_engine/commands/test_move_relative.py [hunk 2]
```diff
@@ -26,7 +26,7 @@ async def test_move_relative_implementation(
 
     result = await subject.execute(data)
 
-    assert result == MoveRelativeResult()
+    assert result == MoveRelativeResult(position=result.position)
```
Comment (mcous) [inline api/tests/opentrons/protocol_engine/commands/test_move_relative.py]: This... doesn't test anything 🙃. It needs to stub the return value of `movement.move_relative`, or else there's nothing to put in the result

**Review by mcous — COMMENTED** [review]
Spotted these issues off the bat in the api side of things. Will fix them and see where that leaves us in terms of the robot-server integration tests

File: app/src/organisms/LabwarePositionCheck/LabwarePositionCheckStepDetail.tsx [hunk 3]
```diff
@@ -36,22 +43,45 @@ import type {
...
   const [showJogControls, setShowJogControls] = React.useState<boolean>(false)
   const { labwareId } = selectedStep
+  const existingOffset = useLabwareOffsetForLabware(runId, labwareId)
+  const [
+    livePositionDeckCoords,
+    setLivePositionDeckCoords,
+  ] = React.useState<Coordinates | null>(null)
+
+  const initialSavePositionCommandId = (savePositionCommandData[labwareId] ??
+    [])[0]
```
Comment (shlokamin) [inline app/src/organisms/LabwarePositionCheck/LabwarePositionCheckStepDetail.tsx:75]: cheeky

File: app/src/organisms/LabwarePositionCheck/LabwarePositionCheckStepDetail.tsx [hunk 4]
```diff
@@ -36,22 +43,45 @@ import type {
...
+  const initialSavePositionCommand = useCommandQuery(
+    runId,
+    initialSavePositionCommandId
+  )?.data?.data
+  const initialPosition =
+    initialSavePositionCommand?.commandType === 'savePosition'
```
Comment (shlokamin) [inline app/src/organisms/LabwarePositionCheck/LabwarePositionCheckStepDetail.tsx:81]: annoying you have to type guard this but i think its safer this way

**Review by shlokamin — APPROVED** [review]
LPC changes LGTM, thank you for the fixes along the way too!

**Pull request merged**